Sources/ObservableArray.swift

```diff
@@ -322,26 +262,24 @@ extension MutableObservableArray where Item: Equatable {
   public func replace(with array: [Item], performDiff: Bool) {
     if performDiff {
       lock.lock()
-      let diff = Array.diff(self.array, array)
-
-      var deletes: [Int] = []
-      var inserts: [Int] = []
-      deletes.reserveCapacity(diff.count)
-      inserts.reserveCapacity(diff.count)
-
-      for diffStep in diff {
-        switch diffStep {
-        case .insert(_, let index):
-          inserts.append(index)
-        case .delete(_, let index):
-          deletes.append(index)
-        }
-      }
```

Comment:
No, no need for manual collation. Do exactly as you did, but just revert

```swift
case insert(Int)
case delete(Int)
case update(Int)
```
back to

```swift
case inserts([Int])
case deletes([Int])
case updates([Int])
```

Does that make sense?